Dockerfile

```diff
@@ -3,6 +3,8 @@ LABEL maintainer="SkynetLabs <devs@skynetlabs.com>"
 
 WORKDIR /root
 
+ENV CGO_ENABLED=0
+
 COPY api api
```

Comment:
This is not a great approach to copying source. The moment we add a new source directory we'll need to change this. And we won't notice this at first because all of the tests will pass without issue, we will only notice it when we try to deploy it on a server.

Let's just copy the entire current directory - that will be a future-proof way of doing it.
What I did some years ago while playing with this was:
```dockerfile
FROM golang:1.13-alpine
ENV CGO_ENABLED 0
ADD  . /go/src/scratch-container/
RUN go test ./... && go install ./...

FROM scratch
COPY --from=0 /go/bin/scratch-container /usr/local/bin/scratch-container
ENTRYPOINT ["scratch-container"]
```

Reply:
> This is not a great approach to copying source. The moment we add a new source directory we'll need to change this. And we won't notice this at first because all of the tests will pass without issue, we will only notice it when we try to deploy it on a server.
> 
> Let's just copy the entire current directory - that will be a future-proof way of doing it. What I did some years ago while playing with this was:
> 
> ```dockerfile
> FROM golang:1.13-alpine
> ENV CGO_ENABLED 0
> ADD  . /go/src/scratch-container/
> RUN go test ./... && go install ./...
> 
> FROM scratch
> COPY --from=0 /go/bin/scratch-container /usr/local/bin/scratch-container
> ENTRYPOINT ["scratch-container"]
> ```

I totally agree that it would be beneficial if all sources would go to src directory but I didn't want to mangle with your codebase. We can do that in the follow up though.

Reply:
There is no need for any mangling. All we need to do is replace the multiple `COPY`s with `ADD  . /go/src/blocker/` and then do a `RUN go test ./... && go install ./...`.

Reply:
The whole thing can look like this:
```dockerfile
FROM golang:1.17.7 as builder
LABEL maintainer="SkynetLabs <devs@skynetlabs.com>"

WORKDIR /blocker

ENV CGO_ENABLED=0
ADD . .
RUN go mod download && make release

FROM alpine:3.15.0
LABEL maintainer="SkynetLabs <devs@skynetlabs.com>"

COPY --from=builder /go/bin/blocker /usr/bin/blocker

ENTRYPOINT ["/usr/bin/blocker"]
```

Reply:
> There is no need for any mangling. All we need to do is replace the multiple `COPY`s with `ADD . /go/src/blocker/` and then do a `RUN go test ./... && go install ./...`.

oh that's what you mean, I usually try to avoid adding unnecessary files and directories and thought that you might want to move all those directories to a single src directory in the repo instead and then just COPY that src directory over

Reply:
1 code snippet == 1000 words

Reply:
> 1 code snippet == 1000 words

I'll try that and if that works I'll commit it 👍